setup.py

```diff
@@ -31,122 +31,124 @@
-      return [ ]
-    else:
-      # Avoids many "type-punned pointer" warnings
-      return [ "-Wno-strict-aliasing" ]
-
-  def link_args(self, mod):
     """
-    Gets additional linker arguments
+    Provides some configuration-specific methods
     """
-    if platform.system() == "Windows":
-      return [ ]
-    elif platform.system() == "Darwin":
-      # For the dependency modules, make sure we produce a dylib.
-      # We can only link against such, but the bundles produced otherwise.
-      if mod[0:1] == "_":
-        return [ "-Wl,-dylib" ]
-      else:
-        return []
-    else:
-      # this makes the libraries suitable for linking with a path -
-      # i.e. from path_of('_tl'). Without this option, the path
-      # will be included in the reference and at runtime the loaded
-      # will look for the path-qualified library. But that's the 
-      # build path and the loader will fail.
-      return ['-Wl,-soname,' + self.libname_of(mod)]
-
-  def macros(self):
-    """
-    Returns the macros to use for building
-    """
-    return [ ('HAVE_CURL', 1), ('HAVE_EXPAT', 1) ]
 
-  def version(self):
-    """
-    Gets the version string
-    """
-    return "0.26"
+    def __init__(self):
+
+        # TODO: is this really how we get the build paths?
+
+        build_cmd = Distribution().get_command_obj('build')
+        build_cmd.finalize_options()
+        self.build_platlib = build_cmd.build_platlib
+
+        install_cmd = Distribution().get_command_obj('install')
+        install_cmd.finalize_options()
+        self.install_platlib = install_cmd.install_platlib
+
+        self.ext_suffix = sysconfig.get_config_var("EXT_SUFFIX")
+        self.root = "klayout"
+
+    def libname_of(self, mod):
+        """
+        Returns the library name for a given module
+        The library name is usually decorated (i.e. "tl" -> "tl.cpython-35m-x86_64-linux-gnu.so").
+        """
+        return mod + self.ext_suffix
+
+    def path_of(self, mod):
+        """
+        Returns the build path of the library for a given module
+        """
+        return os.path.join(self.build_platlib, self.root, self.libname_of(mod))
+
+    def rpath(self):
+        """
+        Returns the runtime_library_dir to use when linking the modules
+        This path will ensure the auxiliary modules are found at runtime.
+        """
+        return [os.path.join(self.install_platlib, self.root)]
+
+    def compile_args(self, mod):
+        """
+        Gets additional compiler arguments
+        """
+        if platform.system() == "Windows":
+            return []
+        elif platform.system() == "Darwin":
+            return []
+        else:
+            # Avoids many "type-punned pointer" warnings
+            return ["-Wno-strict-aliasing"]
+
+    def link_args(self, mod):
+        """
+        Gets additional linker arguments
+        """
+        if platform.system() == "Windows":
+            return []
+        elif platform.system() == "Darwin":
```

Comment:
The changes only affect macosx. Was this already working for linux? If not, maybe we could wait to merge until linux is also working, because I think it is very close.